vllm/compilation/decorators.py

```diff
@@ -222,6 +232,35 @@ def __call__(self, *args, **kwargs):
         if self.do_not_compile or torch.compiler.is_compiling():
             return self.forward(*args, **kwargs)
 
+        if getattr(self, "aot_compiled_fn", None) is not None:
+            return self.aot_compiled_fn(self, *args, **kwargs)
+
+        cache_dir = None
+        aot_compilation_path = None
+        if envs.VLLM_USE_AOT_COMPILE:
+            hash_key = _model_hash_key(self.forward)
+            cache_dir = os.path.join(
+                envs.VLLM_CACHE_ROOT,
+                "aot_compilation",
+                hash_key,
+            )
+
+            rank = self.vllm_config.parallel_config.rank
+            dp_rank = self.vllm_config.parallel_config.data_parallel_rank
+            cache_dir = os.path.join(cache_dir, f"rank_{rank}_{dp_rank}")
+            aot_compilation_path = os.path.join(cache_dir, "model")
+            try:
+                with open(aot_compilation_path, "rb") as f:
+                    aot_compiled_fn = torch.compiler.load_compiled_function(f)
```

Comment:
Is there an internal verification on cuda version, hardware in torch.compile loading?

Reply:
For now we haven't implemented it yet, since this feature is relatively new (why we start with opt-in). Here is a list of guards we plan to implement for loading:
- torch version
- python version
- cuda version
- hardware
- traced source files
Let me know if there's anything I'm missing, thanks.